pygmt/tests/test_subplot.py

```diff
@@ -0,0 +1,110 @@
+"""
```

Comment:
I'm afraid some of the tests may always pass even though we make some backward-incompatible changes. We can explore how to write better and robust tests later.

Reply:
Yeah, definitely not my best attempt as it's mostly testing the aliases and a few subtle quirks. There are probably more complicated tests we can write (especially around `fig.set_panel`), but probably best to stress test things in the real world and make changes as we go along.